cpu/esp32/freertos/semphr.c

```diff
@@ -94,7 +94,7 @@ BaseType_t xSemaphoreTake (SemaphoreHandle_t xSemaphore,
         case queueQUEUE_TYPE_MUTEX:
         {
             if (xTicksToWait == 0) {
-                return (mutex_trylock(mutex) == 0) ? pdTRUE : pdFALSE;
+                return (mutex_trylock(mutex) == 1) ? pdPASS : pdFAIL;
```

Comment:
@JulianHolzwarth Don't change the return values. There is no reason for it. According to [API reference](http://www.freertos.org/a00122.html), the return value of xSemaphoreTake is `pdTRUE` and `pdFALSE`.  Even though the values are the same, we should use them as specified to avoid misleadings in future implementations of the FreeRTOS adaptation layer.

Reply:
I was looking at the Reference Manual. It gave me more information about the function.
https://www.freertos.org/wp-content/uploads/2018/07/FreeRTOS_Reference_Manual_V10.0.0.pdf

Linked from here:
https://www.freertos.org/Documentation/RTOS_book.html

If you like `PdTrue` more I will change it.

Reply:
The  latest official release http://www.freertos.org/a00104.html has `return pdPASS;`
and
https://github.com/aws/amazon-freertos/blob/edc41fb223b8f9529162651a7c0459b10b450164/freertos_kernel/queue.c#L1414
has the same.
(The other return is `return errQUEUE_EMPTY`)

Reply:
That's strange. Obviously, the official offline documentation does not fit the [official online documentation](http://www.freertos.org/a00106.html). And even more strange, the documentation in [semphr.h](https://github.com/aws/amazon-freertos/blob/edc41fb223b8f9529162651a7c0459b10b450164/freertos_kernel/include/semphr.h#L323) does not fit the implementatoin in [queue.c](https://github.com/aws/amazon-freertos/blob/edc41fb223b8f9529162651a7c0459b10b450164/freertos_kernel/queue.c#L1414).

The question is how `errQUEUE_EMPTY` is defined.

Reply:
This might become a pitfall. By default these are defined:
```
#define errQUEUE_EMPTY ( ( BaseType_t ) 0 )
#define pdFALSE        ( ( BaseType_t ) 0 )
#define pdFAIL         ( pdFALSE )
```
Defining something like an error code which is nothing else or the same as `pdFALSE` is quite strange, especially since other error codes are real values
```
#define errQUEUE_BLOCKED    ( -4 )
#define errQUEUE_YIELD      ( -5 )
```
If someone concludes from the online documentation that `xSemaphoreTake` returns something like a boolean, he could try the following:
```
if (!xSemaphoreTake(...)) {
    ...
}
```
Although it would work for `errQUEUE_EMPTY` but not for the other error codes.